shell/app/org-home/stores/org.tsx

```diff
@@ -80,6 +80,10 @@ const org = createStore({
       } else {
         const currentOrg = resOrg || {};
         const orgId = currentOrg.id;
+        // user doesn't joined the public org, go to workBench
+        if (resOrg.isPublic && !(orgs?.list || []).find((x: Obj) => x.name === currentOrg.name)) {
```

Comment:
1. orgs should have type, so no need `x: Obj`
2. `orgs?.list?.find` instead of `orgs?.list || []`